src/SleepRepo.js

```diff
@@ -19,43 +19,41 @@ class SleepRepo {
       }
       return arr;
     }, []);
+    return dataByUser
+  }
 
-    let avgSleepQualityPerUser = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((totalQuality, day) => {
-      totalQuality += day.sleepQuality;
-      return totalQuality;
-    }, 0)).map(user => Number((user / 7).toFixed(2)));
+  returnUsersWeeklyTotals(week, metric) {
+    let dataByUser = this.organizeByUser();
+    let usersWeeklyTotals = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((total, day) => {
+      total += day[metric];
+      return total;
+    }, 0))
+    return usersWeeklyTotals
+  }
 
+  returnWeeksBestSleepQuality(week, metric) {
+    let weekTotals = this.returnUsersWeeklyTotals(week, metric);
+    let weekAverages = weekTotals.map(user => Number((user / 7).toFixed(2)));
     let goodSleepers = [];
-    avgSleepQualityPerUser.forEach((user, index) => {
+    weekAverages.forEach((user, index) => {
       if (user >= 3) {
         goodSleepers.push(index + 1);
       }
     });
-    return goodSleepers;
+    return goodSleepers
+  }
+
+  returnWeeklyLongestSleepers(week, metric) {
+    let avgSleepHoursPerUser = this.returnUsersWeeklyTotals(week, metric)
+    return [Math.max(...avgSleepHoursPerUser), avgSleepHoursPerUser.indexOf(Math.max(...avgSleepHoursPerUser)) + 1];
   }
 
+
   returnLongestSleepers(date) {
```

Comment:
This method wasn't modified in your PR but looking at it - can we rename to reflect longest Sleeper by date?  It will only ever return one user ID.